src/main/java/org/fcrepo/migration/handlers/BasicObjectVersionHandler.java

```diff
@@ -175,12 +162,210 @@ public void processObjectVersions(Iterable<ObjectVersionReference> versions) {
     }
 
     private FedoraObject createObject(ObjectReference object) throws FedoraException {
         return repo.createObject(idMapper.mapObjectPath(object));
     }
 
+    private void updateObjectProperties(ObjectVersionReference version,
+                                        FedoraObject object,
+                                        QuadAcc triplesToRemove,
+                                        QuadDataAcc triplesToInsert) {
+        if (version.isLastVersion()) {
+            for (ObjectProperty p : version.getObjectProperties().listProperties()) {
+                String pred = p.getName();
+                String obj = p.getValue();
+
+                // Map dates and object state
+                if (pred.equals("info:fedora/fedora-system:def/model#createdDate")) {
+                    pred = "http://www.loc.gov/premis/rdf/v1#hasDateCreatedByApplication";
+                } else if (pred.equals("info:fedora/fedora-system:def/model#state")) {
+                    pred = "http://fedora.info/definitions/1/0/access/objState";
+                } else if (pred.equals("info:fedora/fedora-system:def/view#lastModifiedDate")) {
+                    pred = "http://www.loc.gov/premis/rdf/v1#hasEventDateTime";
+
+                    // Add the migration event type for the last modified date
+                    updateTriple(triplesToRemove,
+                                 triplesToInsert,
+                                 "http://www.loc.gov/premis/rdf/v1#hasEventType",
+                                 "migration");
+                }
+
+                if (isDateProperty(p.getName())) {
+                    updateDateTriple(triplesToRemove,
+                                     triplesToInsert,
+                                     pred,
+                                     obj);
+                }
+                else {
+                    updateTriple(triplesToRemove,
+                                 triplesToInsert,
+                                 pred,
+                                 obj);
+                }
+            }
+        }
+
+        // Update if there's triples to remove / add.
+        // Some may come from other datastreams like RELS-EXT and DC, not just
+        // in this function.
+        if (!triplesToInsert.getQuads().isEmpty() && !triplesToRemove.getQuads().isEmpty()) {
+            updateResourceProperties(object, triplesToRemove, triplesToInsert);
+        }
+    }
+
+    /**
+     * WIP utility function to update datastream properties.
+     *
+     * @param v     Version of the datasream to update.
+     * @param ds    Datastream to update. 
+     * @return void 
+     */
+    private void updateDatastreamProperties(DatastreamVersion v, FedoraDatastream ds) {
+        QuadDataAcc triplesToInsert = new QuadDataAcc();
+        QuadAcc triplesToRemove = new QuadAcc();
+
+        String createdDate = v.getCreated();
+
+        if (v.isFirstVersion()) {
```

Comment:
You could provide a boolean "isFirstVersion" to this method that is computed on line 149 with:

version.getObject().getDatastreamVersions(v.getDatastreamInfo().getDatastreamId()).indexOf(v);